### ISSUE TITLE
no matches for kind "MXJob" in version "kubeflow.org/v1beta1"

### ISSUE THREAD

**Issue by CharlesxrWu** [kubeflow/kubeflow#2797]
I can ran the tf-job, but cannot run the mxnet-job.

The command I use to list crd is 
`kubectl get crd`

The output of crd is
`applications.app.k8s.io                      2019-03-18T17:54:51Z`
`...`
`mxjobs.kubeflow.org                          2019-03-24T23:20:39Z`
`pytorchjobs.kubeflow.org                     2019-03-18T17:53:18Z`
`scheduledworkflows.kubeflow.org              2019-03-18T17:53:43Z`
`tfjobs.kubeflow.org                          2019-03-18T21:21:14Z`
`workflows.argoproj.io                        2019-03-18T17:53:36Z`

The command I use to launch the task is 
`kubectl create -f mxnet-operator/examples/v1beta1/train/mx_job_dist_gpu.yaml`

The output is 
`error: unable to recognize "mxnet-operator/examples/v1beta1/train/mx_job_dist_gpu.yaml": no matches for kind "MXJob" in version "kubeflow.org/v1beta1"`

I really appreciate any help.

**Comment by pdmack** [comment]
The MXJob operator and CRD were just recently updated to v1beta1:

https://github.com/kubeflow/mxnet-operator/commit/f1df1f65aaed128300783f7f3475f1d60c8620cb

Try reinstalling your MXJobs operator and CRD as described [here](https://github.com/kubeflow/mxnet-operator#verify-that-mxnet-support-is-included-in-your-kubeflow-deployment).

**Comment by CharlesxrWu** [comment]
Follow the guide
`ks pkg install kubeflow/mxnet-job`

But I cannot install kubeflow/mxnet-job
`ERROR resolve registry library: GET https://api.github.com/repos/kubeflow/kubeflow/contents/kubeflow/mxnet-job?ref=b497fd581c823ba3911296ec58c072a48c57c474: 404 Not Found []`

**Comment by jlewi** [comment]
Code should be there.
https://github.com/kubeflow/kubeflow/tree/master/kubeflow/mxnet-job

@suleisl2000 It looks like you are the primary author of mxnet job. Can you help out here?

**Comment by suleisl2000** [comment]
@jlewi @CharlesxrWu The issue is caused by that mxnet-operator was updated recently due to [kubeflow/mxnet-operator@f1df1f6](https://github.com/kubeflow/mxnet-operator/commit/f1df1f65aaed128300783f7f3475f1d60c8620cb), but I didn't update kubeflow repo in time. I am working on it now. 

@CharlesxrWu As a temporary solution, you can start mxnet-operator manually as below
```bash
kubectl create -f manifests/crd-v1beta1.yaml
kubectl create -f manifests/rbac.yaml
kubectl create -f manifests/deployment.yaml
```

**Comment by suleisl2000** [comment]
@jlewi @pdmack Hi, I fixed the issue at kubeflow#2909 yesterday. Do I need merge the fix to kubeflow:v0.5-branch? I notice that v0.5.0 RC2 is released.

**Comment by pdmack** [comment]
#2909 has not been cherry-picked to the v0.5 branch as of yet. @lluunn ? @jlewi ?

**Comment by jlewi** [comment]
@suleisl2000 Please create a cherry pick for #2909 if one is needed. You can use this script
https://github.com/kubeflow/kubeflow/blob/master/hack/cherry_pick_pull.sh

**Comment by CharlesxrWu** [comment]
Sorry, there is still same error.

I used the latest Kubeflow
`
VERSION=master
ks registry add kubeflow github.com/kubeflow/kubeflow/tree/${VERSION}/kubeflow
`
and tried
`
https://github.com/kubeflow/kubeflow/tree/master/kubeflow/mxnet-job 

#or

https://github.com/kubeflow/mxnet-operator
`
but still
`
error: unable to recognize "test.yaml": no matches for kind "MXJob" in version "kubeflow.org/v1beta1"
`

**Comment by suleisl2000** [comment]
@CharlesxrWu Could you append detail reproduce steps for trouble shooting? And where I can find test.yaml?

**Comment by CharlesxrWu** [comment]
After moving 'version=master' to 'version=v0.5.0', it can create the job.

However, the demo can not work.

I list all commands I use
`
minikube start --cpus 4 --memory 8096 --disk-size=40g
NAMESPACE=mykubeflow
kubectl create ns ${NAMESPACE}
kubectl config set-context $(kubectl config current-context) --namespace=${NAMESPACE}
ks init mykubeflow-app
cd mykubeflow-app
ks env set default --namespace ${NAMESPACE}
VERSION=v0.5.0
ks registry add kubeflow github.com/kubeflow/kubeflow/tree/${VERSION}/kubeflow
ks pkg install kubeflow/mxnet-job
ks generate mxnet-operator mxnet-operator
ks apply default -c mxnet-operator
kubectl create -f examples/v1beta1/train/mx_job_dist_gpu.yaml
delete GPU requirement in yaml
`

Job information:

`apiVersion: v1
items:
- apiVersion: kubeflow.org/v1beta1
  kind: MXJob
  metadata:
    clusterName: ""
    creationTimestamp: "2019-05-08T00:28:04Z"
    generation: 1
    name: mxnet-job
    namespace: mykubeflow
    resourceVersion: "4485"
    selfLink: /apis/kubeflow.org/v1beta1/namespaces/mykubeflow/mxjobs/mxnet-job
    uid: 25040574-7128-11e9-b453-0800271184ad
  spec:
    cleanPodPolicy: All
    jobMode: MXTrain
    mxReplicaSpecs:
      Scheduler:
        replicas: 1
        restartPolicy: Never
        template:
          metadata:
            creationTimestamp: null
          spec:
            containers:
            - image: mxjob/mxnet:gpu
              name: mxnet
              ports:
              - containerPort: 9091
                name: mxjob-port
              resources: {}
      Server:
        replicas: 1
        restartPolicy: Never
        template:
          metadata:
            creationTimestamp: null
          spec:
            containers:
            - image: mxjob/mxnet:gpu
              name: mxnet
              ports:
              - containerPort: 9091
                name: mxjob-port
              resources: {}
      Worker:
        replicas: 1
        restartPolicy: Never
        template:
          metadata:
            creationTimestamp: null
          spec:
            containers:
            - args:
              - /incubator-mxnet/example/image-classification/train_mnist.py
              - --num-epochs
              - "10"
              - --num-layers
              - "2"
              - --kv-store
              - dist_device_sync
              command:
              - python
              image: mxjob/mxnet:gpu
              name: mxnet
              ports:
              - containerPort: 9091
                name: mxjob-port
              resources: {}
  status:
    completionTime: "2019-05-08T00:28:06Z"
    conditions:
    - lastTransitionTime: "2019-05-08T00:28:04Z"
      lastUpdateTime: "2019-05-08T00:28:04Z"
      message: MXJob mxnet-job is created.
      reason: MXJobCreated
      status: "True"
      type: Created
    - lastTransitionTime: "2019-05-08T00:28:04Z"
      lastUpdateTime: "2019-05-08T00:28:06Z"
      message: MXJob mxnet-job is failed.
      reason: MXJobFailed
      status: "True"
      type: Failed
    mxReplicaStatuses:
      Scheduler: {}
      Server: {}
      Worker: {}
`

Worker Information:

`
Name:                      mxnet-job-worker-0
Namespace:                 mykubeflow
Node:                      minikube/10.0.2.15
Start Time:                Wed, 08 May 2019 08:28:04 +0800
Labels:                    group_name=kubeflow.org
                           mxnet-replica-index=0
                           mxnet-replica-type=worker
                           mxnet_job_name=mxnet-job
Annotations:               <none>
Status:                    Terminating (lasts 15s)
Termination Grace Period:  30s
IP:                        172.17.0.7
Controlled By:             MXJob/mxnet-job
Containers:
  mxnet:
    Container ID:  docker://babaf2a9b4f1ed93bb5099c67b004055ce0ef8da0d3675d570ef92184a9a5a3b
    Image:         mxjob/mxnet:gpu
    Image ID:      docker-pullable://mxjob/mxnet@sha256:f0ab7315578dbcddab6af926855d2586190f4f0c3dd5f4bb34f28a5a15ac7c84
    Port:          9091/TCP
    Host Port:     0/TCP
    Command:
      python
    Args:
      /incubator-mxnet/example/image-classification/train_mnist.py
      --num-epochs
      10
      --num-layers
      2
      --kv-store
      dist_device_sync
    State:          Terminated
      Reason:       Error
      Exit Code:    1
      Started:      Wed, 08 May 2019 08:28:05 +0800
      Finished:     Wed, 08 May 2019 08:28:05 +0800
    Ready:          False
    Restart Count:  0
    Environment:
      MX_CONFIG:            {"cluster":{"scheduler":[{"url":"mxnet-job-scheduler-0","port":9091}],"server":[{"url":"mxnet-job-server-0","port":9091}],"worker":[{"url":"mxnet-job-worker-0","port":9091}]},"labels":{"scheduler":"","server":"","worker":""},"task":{"type":"worker","index":0}}
      DMLC_PS_ROOT_PORT:    9091
      DMLC_PS_ROOT_URI:     mxnet-job-scheduler-0
      DMLC_NUM_SERVER:      1
      DMLC_NUM_WORKER:      1
      DMLC_ROLE:            worker
      DMLC_USE_KUBERNETES:  1
    Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-nj9pf (ro)
Conditions:
  Type           Status
  Initialized    True
  Ready          False
  PodScheduled   True
Volumes:
  default-token-nj9pf:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  default-token-nj9pf
    Optional:    false
QoS Class:       BestEffort
Node-Selectors:  <none>
Tolerations:     node.kubernetes.io/not-ready:NoExecute for 300s
                 node.kubernetes.io/unreachable:NoExecute for 300s
Events:
  Type    Reason                 Age   From               Message
  ----    ------                 ----  ----               -------
  Normal  Scheduled              16s   default-scheduler  Successfully assigned mxnet-job-worker-0 to minikube
  Normal  SuccessfulMountVolume  16s   kubelet, minikube  MountVolume.SetUp succeeded for volume "default-token-nj9pf"
  Normal  Pulled                 15s   kubelet, minikube  Container image "mxjob/mxnet:gpu" already present on machine
  Normal  Created                15s   kubelet, minikube  Created container
  Normal  Started                15s   kubelet, minikube  Started container
`
